---
title: "Mutations in legume symbiosis genes create a complex selective landscape for natural genetic variation in rhizobia during multi-strain inoculations"
author: Sohini Guha,Gina Bledsoe,Liana T. Burghardt
date: "`r Sys.Date()`"
output: html_document
---

Analysis file for "Mutations in legume symbiosis genes create a complex selective landscape for natural genetic variation in rhizobia during multi-strain inoculations"

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)


#Load packages needed. Tidyverse include dplyer
require(tidyverse)
require(ggpubr)
require(ggplot2)
require(RColorBrewer)
require(gplots)
require(vegan)
require(corrplot)
knitr::opts_chunk$set(message = FALSE)
```

###Compiling the data for analysis

```{r Code for creating FigS1, echo=TRUE}
# Strain frequency files - freq.tsv
freq_all<-read.table("../data/freq.tsv", header=TRUE)
genotype_name <- read.table("../data/genotype_name.txt",sep = "\t",header=TRUE)

genotype_name<-as_tibble(genotype_name)%>% mutate(across(!order, as.factor)) %>% filter(genotype_id!="838")

# Subset initial inoculum replications and calculate mean fitness across technical replicates
freq_initial<- freq_all %>% filter(str_detect(sample_id, "M_C86_NL")) %>% select(!sample_id) %>% summarise_all(mean)

# Graph the initial frequency histogram
hist(t(freq_initial),breaks = 10, main= "Initial strain frequency distribution")

pdf(file = "../figures/FigS1.pdf",width = 6, height = 6)
hist(t(freq_initial),breaks = 10, main= "Initial strain frequency distribution", xlab="Isolate Frequency", ylab="Number of Isolates")
dev.off()

#Break apart sample_id string into genotype id and replicate and convert into factors
freq_all<-freq_all %>% mutate(genotype_id = substr(sample_id, 3,5), rep = substr(sample_id, 6,6),.before = sample_id) %>% mutate(across(c(genotype_id,rep,sample_id), as.factor))
                                                                                      
 #join the dataset of genotype names with the genotype ids
freq_all <- inner_join(genotype_name,freq_all, by="genotype_id") %>% filter(background=="A17") %>% rename(geno=name)

freq_all$type<-factor(freq_all$type, levels = c("WT","Early","Late","AON"))

freq_all$geno<-factor(freq_all$geno,levels=c("A17", "dmi1-2","dmi2-3","dmi3-1","hcl","ipd3","latd", "nad1", "dnf1", "dnf2","dnf3", "dnf4","dnf6","dnf7", "dnf1/dnf2", "dnf5/dnf2", "rdn1", "sunn1","sunn4"))

#Transform raw frequency data to fitness data= log2(treatment_freq/initial_freq)
fit_all<-as_tibble(cbind(freq_all[,c(1:7)],log2(mapply("/", freq_all[,c(-1:-7)], freq_initial))))

#The lowest fitness values is Mag27 = -7.1 to substitute all INF with -8
fit_all<-fit_all %>% mutate_if(is.numeric, ~ifelse(abs(.) == Inf,-8,.))

write.table (fit_all,file="../data/Strain Fitness_MedicagoMutants_A17.txt",sep = "\t",row.names = FALSE)

fit_all$geno<-factor(fit_all$geno)
# Create a data frame with fitness means for each geno                            
fit_mean<-fit_all %>% group_by(geno,type) %>% summarise(across(starts_with("MAG"),mean))
fit_median<-fit_all %>% group_by(geno,type) %>% summarise(across(starts_with("MAG"),median))

# Calculate strain diversity and create a new dataset
div_all<-rowwise(freq_all) %>% transmute(geno=geno,type=type, rep=rep,sample_id=substr(sample_id, 3,6), nodule_isolate_diversity=diversity(c_across(starts_with("MAG"))))

```

```{r Merge all trait data sets, echo=TRUE}
#Names of the Medicago genotypes in A17 background along with their genotype ids, disruption type and order-genotype_name.txt
#Nodule numbers and individual nodule sizes for each sample- data_HostMutants2021_nod_counts.txt
#Sample ids, replicates for each genotype,CFU per ml,aboveground biomass,belowground biomass for each sample-data_nsfpgrp_hostmutants_C86meliloti.txt

# Import the files
data_HostMutants2021_nod_counts <- read.table("../data/data_HostMutants2021_nod_counts.txt",sep = "\t",header=TRUE)

data_nsfpgrp_hostmutants_C86meliloti <- read.table("../data/data_nsfpgrp_hostmutants_C86meliloti.txt",sep = "\t",header=TRUE)
# Merge the datasets sequentially
traits_all <-as_tibble(full_join (data_HostMutants2021_nod_counts,data_nsfpgrp_hostmutants_C86meliloti,by=c("sample_id","genotype_id","rep"))) %>% mutate(across(c(genotype_id,rep),as.factor))

traits_all <-full_join (genotype_name,traits_all,by = "genotype_id") %>% filter(background=="A17")%>% rename(geno=name)

traits_all <- full_join(traits_all,div_all,by=NULL)

###### Data set clean-up ######
# Choose only columns to keep that are meaningful for our analysis
traits_all<-subset (traits_all,select=c(sample_id,genotype_id,geno,type,rep,background,date_biomass_harvested,num_plants,percent_herbivory,flowering,above_biomass_g,below_biomass_g,Count,num_plants_picked_nods,total_area,average_size,cfu_ml,nodule_isolate_diversity))

# Calculate additional measurements
#Rename the traits
traits_all<-traits_all%>% mutate(
  above_biomass_per_plant=above_biomass_g/num_plants,
  below_biomass_per_plant=below_biomass_g/num_plants,
  root_to_shoot_ratio=below_biomass_per_plant/above_biomass_per_plant,
  total_biomass_per_plant=above_biomass_per_plant+below_biomass_per_plant,
  nodule_number_per_plant=Count/num_plants_picked_nods,
  cfu_per_plant_LOG10=log10(cfu_ml/num_plants_picked_nods),
  cfu_per_nodule_LOG10=log10(cfu_ml/Count),
  total_nodulated_area_per_plant=total_area/num_plants_picked_nods,
  average_area_per_nodule=total_nodulated_area_per_plant/nodule_number_per_plant)

#Replacing  'NAN' and 'Inf' with 'NA'
traits_all[is.na(traits_all) | traits_all == "Inf"] <- NA
traits_all[is.na(traits_all) | traits_all == "NaN"] <- NA

#Turning traits_all$type into a factor.
traits_all$type<-factor(traits_all$type, levels = c("WT","Early","Late","AON"))
#Changing the order of the genotypes
traits_all$geno<-factor(traits_all$geno,levels=c("A17", "dmi1-2","dmi2-3","dmi3-1","hcl","ipd3","latd", "nad1", "dnf1", "dnf2","dnf3", "dnf4","dnf6","dnf7", "dnf1/dnf2", "dnf5/dnf2", "rdn1", "sunn1","sunn4"))
```

### Statistical Analysis: Coeffecients from ANOVA's and RDAs

```{r Perform analogous RDA Permanova on strain fitness data}
rda_geno<-rda(fit_all[,c(8:93)]~geno+rep,fit_all, scale=TRUE)
rda.model.geno<-anova(rda_geno, step=1000, perm.max=1000, by= "terms")  

rda_type<-rda(fit_all[,c(8:93)]~type+rep,fit_all, scale=TRUE)
rda.model.type<-anova(rda_type, step=1000, perm.max=1000, by= "terms") 

rda.strainfitness<-data.frame(Response=c("nodule_isolate_fitness"),Predictor=c("Genotype","Replicate","Residual","Type","Replicate","Residual"),Sum.Sq=c(rda.model.geno$Variance,rda.model.type$Variance),PctExp=c(rda.model.geno$Variance/sum(rda.model.geno$Variance)*100,rda.model.type$Variance/sum(rda.model.type$Variance)*100),Df=c(rda.model.geno$Df,rda.model.type$Df),Fval=c(rda.model.geno$F,rda.model.type$F), Pval=c(rda.model.geno$`Pr(>F)`,rda.model.type$`Pr(>F)`),R.adj=c(round(RsquareAdj(rda_geno)$adj.r.squared,4),"NA","NA",round(RsquareAdj(rda_type)$adj.r.squared,4),"NA","NA")) #do we need to compile this in a table??
```

```{r Write Function to run anova and summarize results each Phenotype, echo=TRUE}

extract.anova<- function(trait) {
#Dropping rows with NA

traits_all<-as.data.frame(traits_all[!is.na(traits_all[,trait]),]) 

# Run a model with genotype and replicate as predictors
NodNum.G <- lm(traits_all[,trait] ~ geno + rep, data = traits_all)
# Run a model with type and replicate as predictors
NodNum.T<-lm(traits_all[,trait] ~ type + rep, data = traits_all)
#Examine the Analysis of Variance tables and save output
a.G<-anova(NodNum.G)
a.T<-anova(NodNum.T)

#Compile results into a dataframe
myresults<-data.frame(Response=c(paste(trait)),Predictor=c("Genotype","Replicate","Residual","Type","Replicate","Residual"),Sum.Sq=c(a.G$"Sum Sq",a.T$"Sum Sq"),PctExp=c(a.G$"Sum Sq"/sum(a.G$"Sum Sq")*100,a.T$"Sum Sq"/sum(a.T$"Sum Sq")*100),Df=c(a.G$"Df",a.T$"Df"),Fval=c(a.G$"F value",a.T$"F value"), Pval=c(a.G$"Pr(>F)",a.T$"Pr(>F)"),R.adj=c(summary(NodNum.G)$adj.r.squared,"NA","NA",summary(NodNum.T)$adj.r.squared,"NA","NA"))

return(myresults)

}

#Compile Anova results across Traits
ANOVA.Trait.Summary<-rbind(extract.anova("nodule_number_per_plant"),extract.anova("average_area_per_nodule"),extract.anova("total_nodulated_area_per_plant"),extract.anova("total_biomass_per_plant"),extract.anova("root_to_shoot_ratio"),extract.anova("cfu_per_nodule_LOG10"),extract.anova("cfu_per_plant_LOG10"),extract.anova("nodule_isolate_diversity"),rda.strainfitness)

write.table (ANOVA.Trait.Summary,file="../tables/TableS2.txt",sep = "\t",row.names = FALSE)
```

```{r Statistical summary on geno coefficients, echo=FALSE}

extract.coefs<-function(trait, predictor="geno", p.threshold=0.05) {

# Step1: Remove rows with NA for the focal trait
  my.data<-as.data.frame(traits_all[!is.na(traits_all[,trait]),]) 

# Step2: Run a lm model with genotypes and replicate as predictors

if(predictor=="geno")  {  
    my.model <- lm(my.data[,trait] ~ geno + rep, data = my.data)
}
  
if(predictor=="type")  {
  my.model<- lm(my.data[,trait] ~ type + rep, data = my.data)
}

  # Step 3 Extract coeffecient dataframe and add column for row.names
  my.coefs <- as.data.frame(summary(my.model)$coefficients) # covert to data fram
  my.coefs <- data.frame (Trait=paste(trait),Predictor=row.names(my.coefs), my.coefs) # add the row names as column
  colnames(my.coefs) <-c("Trait","Predictor","Coef", "St.error","t.value","P.value")# Create more meaningful column names

# Step 4 Remove none geno predictors
  my.coefs <-my.coefs[!my.coefs$Predictor %in% c("(Intercept)","repB","repC","repD","repE","repF"),] 

# Step 5: classify coeffecients
  my.coefs$Class<-"None" #Set up dummy column to hold classifications

# Step 6 Modify catagorizations based on coef and pvalue. 
  try(my.coefs[(my.coefs$Coef > 0 & my.coefs$P.value< p.threshold),]$Class<-"Up") 
  try(my.coefs[(my.coefs$Coef < 0 & my.coefs$P.value< p.threshold),]$Class<-"Down") 

return(my.coefs)
}

Geno.Coef.Summary<-rbind(extract.coefs("nodule_number_per_plant"),extract.coefs("average_area_per_nodule"),extract.coefs("total_nodulated_area_per_plant"),extract.coefs("total_biomass_per_plant"),extract.coefs("root_to_shoot_ratio"),extract.coefs("cfu_per_nodule_LOG10"),extract.coefs("cfu_per_plant_LOG10"),extract.coefs("nodule_isolate_diversity"))



Type.Coef.Summary<-rbind(extract.coefs("nodule_number_per_plant",predictor="type"),extract.coefs("average_area_per_nodule",predictor="type"),extract.coefs("total_nodulated_area_per_plant",predictor="type"),extract.coefs("total_biomass_per_plant",predictor="type"),extract.coefs("root_to_shoot_ratio",predictor="type"),extract.coefs("cfu_per_nodule_LOG10",predictor="type"),extract.coefs("cfu_per_plant_LOG10",predictor="type"),extract.coefs("nodule_isolate_diversity",predictor="type"))

write.table (Type.Coef.Summary,file="../tables/TableS3.txt",sep = "\t",row.names = FALSE)
write.table (Geno.Coef.Summary,file="../tables/TableS4.txt",sep = "\t",row.names = FALSE)
```

```{r RDA Model results for Strain Isolate Fitness}

extract.rda<-function(mygroup, predictor="geno", p.threshold=0.05) {

#Step 1: Remove all genotypes but the focal genotype and WT  

if(predictor=="geno")  {
  fit_sub<-fit_all[fit_all$geno==mygroup|fit_all$geno=="A17",] 
  rda_model<-rda(fit_sub[,c(8:93)]~geno,fit_sub, scale=TRUE)
  rda_anova<-anova(rda_model, step=1000, perm.max=1000, by= "terms") 
}  

if(predictor=="type")  {
  fit_sub<-fit_all[fit_all$type==mygroup|fit_all$type=="WT",] 
  rda_model<-rda(fit_sub[,c(8:93)]~type,fit_sub, scale=TRUE)
  rda_anova<-anova(rda_model, step=1000, perm.max=1000, by= "terms")  
}  

my.coefs <-data.frame(Trait="nodule_isolate_fitness",Predictor=paste(mygroup),R.adj=c(round(RsquareAdj(rda_model)$adj.r.squared,4)),Prop.Var=round(rda_anova[1,"Variance"]/sum(rda_anova$Variance),4), f.value=rda_anova[1,"F"],P.value=round(rda_anova [1,"Pr(>F)"],4),Class="None") 

# Step 6 Modify catagorizations based on coef and pvalue. 
  try(my.coefs[(my.coefs$P.value< p.threshold),]$Class<-"Change") 
  
return(my.coefs)

}  
# Compile results for Genotype RDA's
Geno.Coef.RDA<-rbind(extract.rda("hcl"),extract.rda("ipd3"),extract.rda("latd"),extract.rda("nad1"),extract.rda("dnf1"),extract.rda("dnf2"),extract.rda("dnf3"),extract.rda("dnf4"),extract.rda("dnf6"),extract.rda("dnf7"),extract.rda("dnf1/dnf2"),extract.rda("dnf5/dnf2"),extract.rda("rdn1"),extract.rda("sunn1"),extract.rda("sunn4"))

Type.Coef.RDA<-rbind(extract.rda("AON",predictor="type"),extract.rda("Early",predictor = "type"),extract.rda("Late",predictor = "type"))

write.table (Type.Coef.RDA,file="../tables/TableS5.txt",sep = "\t",row.names = FALSE)
write.table (Geno.Coef.RDA,file="../tables/TableS6.txt",sep = "\t",row.names = FALSE)


```

###Visualizing Results

```{r Code for creating FigS2,echo=TRUE}
# Correlation Plot
pdf(file = "../figures/FigS2.pdf",width = 9, height = 9)

traits_all.2<-traits_all%>% select(-sample_id,-rep,-genotype_id,-background,-date_biomass_harvested,-num_plants,-average_size,-percent_herbivory,-flowering,-num_plants_picked_nods,-above_biomass_g,-type,-below_biomass_g,-cfu_ml,-type,-Count,-total_area,-below_biomass_per_plant,-above_biomass_per_plant,-total_area)%>%group_by(geno)%>%summarise_if(is.numeric, mean, na.rm = TRUE)%>%filter(!geno %in% c("dmi1-2", "dmi2-3", "dmi3-1", "skl"))
#converting it to a matrix
traits_all.2<-as.matrix(traits_all.2[,-1])
#corrplot
correlation_matrix <- cor(traits_all.2, method = "pearson", use = "pairwise.complete.obs")
corrplot.mixed(correlation_matrix, number.cex=1,upper = "ellipse", lower = "number",tl.pos = "lt", tl.col = "black",order = 'hclust', tl.offset=1, tl.srt = 90)
dev.off()
```

```{r Code for creating FigS3, echo=TRUE}
#Fig S4a
Geno.Coef.Summary<-Geno.Coef.Summary %>% mutate(Predictor=substr(Predictor,5,14))
Geno.Coef.Graphs<-rbind(Geno.Coef.Summary[,c(1,2,6,7)], Geno.Coef.RDA[,c(1,2,6,7)])
Geno.Coef.Graphs$Class<-factor(Geno.Coef.Graphs$Class,levels=c("None","Up","Down","Change"))
Geno.Coef.Graphs$Trait<-factor(Geno.Coef.Graphs$Trait,levels=c("nodule_number_per_plant","average_area_per_nodule","total_nodulated_area_per_plant","total_biomass_per_plant","root_to_shoot_ratio", "cfu_per_nodule_LOG10","cfu_per_plant_LOG10","nodule_isolate_diversity", "nodule_isolate_fitness"))

Coef.Prop.Figure<-ggplot(data = Geno.Coef.Graphs) + 
  geom_bar(mapping = aes(x = Trait, fill = Class), position = "fill")+ 
  theme_bw()+
  coord_flip()+
  scale_fill_manual(values=c('#F7F7F7','#B2182B', '#2166AC','darkgoldenrod'))+xlab ("")+ylab("Proportion of Genotypes")+theme_bw()

pdf(file = "../figures/FigS4a.pdf",width = 6, height = 6)
Coef.Prop.Figure
dev.off()

#Fig S4b
Geno.Coef.Graphs$Class<-factor(Geno.Coef.Graphs$Class,levels=c("Down","None","Up","Change"))
Geno_matrix<-Geno.Coef.Graphs %>% select(c(Trait,Predictor,Class)) %>% pivot_wider(names_from=c(Trait),values_from=c(Class)) %>% mutate(across(!Predictor,as.numeric)) %>%
column_to_rownames(var="Predictor")
#Create your palette
pal = colorRampPalette(c('#2166AC', '#F7F7F7', '#B2182B','darkgoldenrod'))(4)



col_letters = c("hcl"="#56B4E9","ipd3"="#56B4E9","latd"="#56B4E9", "nad1"="#009E73", "dnf1"="#009E73", "dnf2"="#009E73","dnf3"="#009E73", "dnf4"="#009E73","dnf6"="#009E73","dnf7"="#009E73", "dnf1/dnf2"="#009E73", "dnf5/dnf2"="#009E73", "rdn1"="#E69F00", "sunn1"="#E69F00","sunn4"="#E69F00","dmm1-2"="#56B4E9","dmi2-3"="#56B4E9","dmi3-1"="#56B4E9")

pdf(file = "../figures/FigS4b.pdf",width = 6, height = 3)
heatmap.2(t(as.matrix(Geno_matrix)),Rowv=FALSE, Colv=FALSE,dendrogram = "none",col=pal, trace="none", scale="none", cexCol = .75,cexRow = .75, key=FALSE,srtCol=45, margins=c(5,8), lwid = c(1,15), lhei = c(1,15),na.color='lightgrey',colCol = col_letters)
dev.off()
```

```{r Code for creating Fig2 and Fig3, echo=TRUE}

cbPalette <- c("#B23214","#56B4E9", "#009E73", "#E69F00")

trait.boxplot<-function(trait,mypal=cbPalette,log=FALSE,trait_all=traits_all) {
  trait_all$geno<-factor(trait_all$geno,levels = rev(levels(trait_all$geno))) # reorder factors
  trait_all<-as.data.frame(trait_all[!is.na(trait_all[,trait]),])
  mydata<-data.frame(geno=trait_all$geno, type=trait_all$type, mytrait=trait_all[,trait]) # create sub dataframe with only focal trait to make plotting easier
  
  ggplot(mydata,aes(x=geno,y=mytrait,fill=type))+
        geom_boxplot(outlier.shape=NA,na.rm = TRUE,coef=0) +
        geom_jitter() +
        coord_flip() +
        scale_fill_manual(values=mypal)  +
        labs(x="Plant Host",y=c(paste(trait))) +
        theme_classic() + 
        theme(legend.position="top", text = element_text(size=18))
 
}

pdf(file = "../figures/Fig2.pdf",width = 6, height = 6)
trait.boxplot(trait = "nodule_number_per_plant")
trait.boxplot(trait = "total_nodulated_area_per_plant")
trait.boxplot(trait = "average_area_per_nodule")
trait.boxplot(trait = "root_to_shoot_ratio")
trait.boxplot(trait = "total_biomass_per_plant")
dev.off()

pdf(file = "../figures/Fig3.pdf",width = 6, height = 6)
trait.boxplot(trait = "cfu_per_nodule_LOG10")
trait.boxplot(trait = "cfu_per_plant_LOG10",trait_all = traits_all[traits_all$geno!="latd",]) ## we are sticking to this???
trait.boxplot(trait = "nodule_isolate_diversity")
dev.off()


```

```{r Code for creating Fig4, echo=TRUE}

pal = colorRampPalette(c('#2166AC',"#89ADD0", '#F7F7F7', '#B2182B'))(length(seq(-7.9, 4, 0.1)))
#pal = colorRampPalette(c('#2166AC','#F7F7F7', '#B2182B'))(length(seq(-7.9, 4, 0.1)))


col_letters = c("A17"="#B23214","hcl"="#56B4E9","ipd3"="#56B4E9","latd"="#56B4E9", "nad1"="#009E73", "dnf1"="#009E73", "dnf2"="#009E73","dnf3"="#009E73", "dnf4"="#009E73","dnf6"="#009E73","dnf7"="#009E73", "dnf1/dnf2"="#009E73", "dnf5/dnf2"="#009E73", "rdn1"="#E69F00", "sunn1"="#E69F00","sunn4"="#E69F00")

fit_mean_matrix<-fit_mean %>% select(!type) %>% column_to_rownames(var="geno")
fit_median_matrix<-fit_median %>% select(!type) %>% column_to_rownames(var="geno")

pdf(file = "../figures/Fig4.pdf",width = 5, height = 10)
heatmap.2(t(as.matrix(fit_mean_matrix)),Rowv=TRUE, Colv=TRUE,dendrogram = "both", col=pal,hclustfun = function(d) hclust(d,method="ward.D2"),trace="none", scale="none",cexCol = 1,cexRow = .5, key=TRUE,key.xlab = "Mean Fitness", key.ylab =NA, tracecol="black",symbreaks=FALSE,key.title = NA,colCol = col_letters) #color key helps me understand the dynamics.

heatmap.2(t(as.matrix(fit_median_matrix)),Rowv=TRUE, Colv=TRUE, dendrogram = "both",col=pal,hclustfun = function(d) hclust(d,method="ward.D2"), trace="none", scale="none", cexCol = 1,cexRow = .5, key=TRUE,key.xlab = "Median Fitness", key.ylab =NA , tracecol="black",symbreaks=FALSE,key.title = NA,colCol = col_letters)

myorder<-order(rank(fit_mean_matrix["A17",]))
heatmap.2(t(as.matrix(fit_mean_matrix[,myorder])),Rowv=FALSE, Colv=TRUE, dendrogram = "column", hclustfun = function(d) hclust(d,method="ward.D2"), col=pal, trace="none", scale="none", cexCol = 1,cexRow = .5, key=TRUE,key.xlab = "Mean Fitness", key.ylab =NA , tracecol="black",symbreaks=FALSE, key.title = NA,colCol = col_letters)

myorder<-order(rank(fit_median_matrix["A17",]))
heatmap.2(t(as.matrix(fit_median_matrix[,myorder])),Rowv=FALSE, Colv=TRUE, dendrogram = "column", hclustfun = function(d) hclust(d,method="ward.D2"),col=pal, trace="none", scale="none", cexCol = 1,cexRow = .5, key=TRUE,key.xlab = "Median Fitness", key.ylab =NA , tracecol="black",symbreaks=FALSE, key.title = NA,colCol = col_letters)


dev.off()

```

```{r Code for creating FigS5,echo=TRUE}

df1  <- data.frame(geno=fit_all$geno,type=fit_all$type,summary(rda_geno)$site)    
rda1_ev <-as.vector(eigenvals(rda_geno, model = c("constrained")))

rdainfo<-data.frame(geno=levels(factor(fit_all$geno)),mycols=c("#B23214","#56B4E9","#56B4E9","#56B4E9","#009E73","#009E73","#009E73","#009E73","#009E73","#009E73","#009E73","#009E73","#009E73","#E69F00","#E69F00","#E69F00"), myshapes=c(19,19,18,17,19,18,17,1,2,3,4,5,6,19,18,17))

pdf(file = "../figures/FigS5.pdf",width = 5, height = 5)

ggplot(df1, aes(x=RDA1,y=RDA2,color=type))+
  scale_color_manual(values=cbPalette)+
  geom_text(aes(label=geno),size=3) +
  geom_hline(yintercept=0, linetype="dotted") +
  geom_vline(xintercept=0, linetype="dotted") +
  labs(x=paste("RDA 1 (",round(rda1_ev[1],2),"%)"),y=paste("RDA 2 (",round(rda1_ev[2],2),"%)"), color="Type") +
  theme_classic() + theme(legend.position="top", text = element_text(size=12))

ggplot(df1, aes(x=RDA3,y=RDA4,color=type))+
  scale_color_manual(values=cbPalette)+
  geom_text(aes(label=geno),size=3) +
  geom_hline(yintercept=0, linetype="dotted") +
  geom_vline(xintercept=0, linetype="dotted") +
  labs(x=paste("RDA 3 (",round(rda1_ev[3],2),"%)"),y=paste("RDA 4 (",round(rda1_ev[4],2),"%)"), color="Type") +
  theme_classic() + theme(legend.position="top", text = element_text(size=12)) 

dev.off()

```

```{R Subset data for GWAS Analysis}

x<- as_tibble(t(fit_mean_matrix)) 
# Choices "latd","ipd3", "nad1","dnf2","dnf3","dnf6","dnf7","sunn4" 

GWAS_traits<-tibble(strains=row.names(t(fit_mean_matrix)),A17=x$A17,latd.change=x$latd-x$A17,ipd3.change=x$ipd3-x$A17,nad1change=x$nad1-x$A17,dnf2.change=x$dnf2-x$A17,dnf3.change=x$dnf3-x$A17,dnf6.change=x$dnf6-x$A17,dnf7.change=x$dnf7-x$A17,sunn4.change=x$sunn4-x$A17,sunn1.change=x$sunn1-x$A17,rdn1.change=x$rdn1-x$A17,hcl.change=x$hcl-x$A17,dnf1.change=x$dnf1-x$A17,dnf4.change=x$dnf4-x$A17,dnf5dnf2.change=x$`dnf5/dnf2`-x$A17,dnf1dnf2.change=x$`dnf1/dnf2`-x$A17)

write.table (GWAS_traits,file="../tables/TraitsforGWAS_mean.txt",sep = "\t",row.names = FALSE)
```

```{bash, engine.opts='-l'}
#### Set requirements ####
salloc -N 1 -n 20 --mem-per-cpu=2048 -t 8:00:00

#First, create the gwas environment to install nessesary packages!!
#conda list -e > spec-file.txt
#conda create --name gwas_vcf --file spec-file.txt

#### Activate environment ####
conda activate gwas_vcf

#### SETTINGS ####
MIN_MAF=0.05
MAX_MISSING=0.2
N_PERMUTATIONS=1000


#### Files needed ####
GENOTYPES="ensifer_88_strains_normalized_for_gwas.vcf.gz" #SNPs file obatined from https://datadryad.org/stash/landing/show?id=doi%3A10.5061%2Fdryad.w3r2280sq
PHENOTYPES='TraitsforGWAS_mean.tsv' # first column strain list, phenotypes begin at column 2, head should include phenotype names (i.e., strain, pheno1, pheno2 ...)
STRAINS='strain_list.txt' #list of strains in phenotypes (i.e. column 1)
LD_SITES='ld_sites.txt' #list of LD representative sites, user generated see below (line 27) # 26918 sites 


ANNOTATION_FOR_VARIANTS="annotated_variants.closest.circular.tsv" # From Dryad or Brendan(?)
chr_var="chr_one_variant.tsv" 
psyma_var="psyma_one_variant.tsv" 
psymb_var="psymb_one_variant.tsv" 

nPHENOS=`awk -F'\t' '{print NF; exit}' $PHENOTYPES`
pheno_names=($(awk 'NR==1{print}' $PHENOTYPES))

#### Used to generate LD snps list ####
# wget -O ensifer_ld_groups.zip https://datadryad.org/stash/downloads/file_stream/1823551
# unzip ensifer_ld_groups.zip
# awk 'NR == 1 || FNR > 1' ensifer_ld_groups/ld_group_assignments* > all_ld_group_assignments.tsv
# awk '$7 == "1"' all_ld_group_assignments.tsv | cut -f 1,2 > ld_sites.txt


#### Filter Genotypes ####

bcftools view --samples-file "$STRAINS" "$GENOTYPES" -o filtered.vcf

#### index for downstream filtering ####
rm -f filtered.vcf.gz
bgzip filtered.vcf && tabix -p vcf filtered.vcf.gz

bcftools query -f '%POS\n' filtered.vcf.gz | wc -l  #254255
bcftools query -f '%CHROM\n' filtered.vcf.gz | uniq


#### Filter SNPs by LD snps ####
bcftools view -R $LD_SITES filtered.vcf.gz | bcftools sort -o subset_LD.vcf
rm -f subset_LD.vcf.gz
bgzip subset_LD.vcf && tabix -p vcf subset_LD.vcf.gz # index for downstream filtering


#### Filter sites based on output/chromosome ####
bcftools view subset_LD.vcf.gz --regions USDA1106-chr,KH35c-chr,T073-chr -o chr_subset_LD.vcf
bcftools view subset_LD.vcf.gz --regions USDA1106-psyma,T073-psyma,KH35c-psyma -o psyma_subset_LD.vcf
bcftools view subset_LD.vcf.gz --regions USDA1106-psymb,T073-psymb,KH35c-psymb -o psymb_subset_LD.vcf

#### Index ####
rm -f chr_subset_LD.vcf.gz
rm -f psyma_subset_LD.vcf.gz
rm -f psymb_subset_LD.vcf.gz

bgzip chr_subset_LD.vcf && tabix -p vcf chr_subset_LD.vcf.gz # index for downstream filtering
bgzip psyma_subset_LD.vcf && tabix -p vcf psyma_subset_LD.vcf.gz # index for downstream filtering
bgzip psymb_subset_LD.vcf && tabix -p vcf psymb_subset_LD.vcf.gz # index for downstream filtering



#### Check filtering ####
#sites
bcftools query -f '%POS\n' ensifer_88_strains_normalized_for_gwas.vcf.gz | wc -l # 254255
bcftools query -f '%POS\n' filtered.vcf.gz | wc -l # 254255 
bcftools query -f '%POS\n' subset_LD.vcf.gz | wc -l # 30076
bcftools query -f '%POS\n' chr_subset_LD.vcf.gz | wc -l # 1962
bcftools query -f '%POS\n' psyma_subset_LD.vcf.gz | wc -l # 11137 
bcftools query -f '%POS\n' psymb_subset_LD.vcf.gz | wc -l # 16929

#genotypes
bcftools query -l ensifer_88_strains_normalized_for_gwas.vcf.gz | wc -l #88
bcftools query -l filtered.vcf.gz | wc -l # 86 
bcftools query -l subset_LD.vcf.gz | wc -l # 86
bcftools query -l chr_subset_LD.vcf.gz | wc -l # 86
bcftools query -l psyma_subset_LD.vcf.gz | wc -l # 86
bcftools query -l psymb_subset_LD.vcf.gz | wc -l # 86


#### Subset phenotypes by strain_list ####
grep -f $STRAINS $PHENOTYPES > phenos_subset.fam # 86 


## Convert vcf to bed for GEMMA
# chr
plink --vcf chr_subset_LD.vcf.gz --make-bed --allow-extra-chr --allow-no-sex --maf 0.05 --geno --pheno "phenos_subset.fam" --all-pheno --out "chr_subset_LD_plink"

# psyma
plink --vcf psyma_subset_LD.vcf.gz --make-bed --allow-extra-chr --allow-no-sex --maf 0.05 --geno --pheno "phenos_subset.fam" --all-pheno --out "psyma_subset_LD_plink"

# psymb
plink --vcf psymb_subset_LD.vcf.gz --make-bed --allow-extra-chr --allow-no-sex --maf 0.05 --geno --pheno "phenos_subset.fam" --all-pheno --out "psymb_subset_LD_plink"

## Relatedness matrix
#"-gk 2" calculates the standardized relatedness matrix
#Note: need to have the .fam file for it to run: e.g., cp subset_LD_plink.fam chr_subset_LD_plink.fam

gemma -bfile chr_subset_LD_plink -p phenos_subset.fam -gk 2 -maf $MIN_MAF -miss $MAX_MISSING -o chr_subset_LD_ksmat
gemma -bfile psyma_subset_LD_plink -p phenos_subset.fam -gk 2 -maf $MIN_MAF -miss $MAX_MISSING -o psyma_subset_LD_ksmat
gemma -bfile psymb_subset_LD_plink -p phenos_subset.fam -gk 2 -maf $MIN_MAF -miss $MAX_MISSING -o psymb_subset_LD_ksmat


#####################################################################################################################################################
## LMM option in GEMMA
#-n <number> specifies column corresponding to phenotype in .fam file.
#-lmm option 4 includes all three stats tests (see GEMMA manual for more details)

## This is the "real" gwas
 for i in `seq 2 $nPHENOS`; # 
 do
  gemma -bfile chr_subset_LD_plink -p phenos_subset.fam -k ./output/chr_subset_LD_ksmat.sXX.txt \
  -maf $MIN_MAF -miss "$MAX_MISSING" \
  -lmm 4 -n $i -o chr_subset_LD_ulmm_trait_$i
  echo "For chr, ran trait_$i through GEMMA"

  gemma -bfile psyma_subset_LD_plink -p phenos_subset.fam -k ./output/psyma_subset_LD_ksmat.sXX.txt \
  -maf $MIN_MAF -miss "$MAX_MISSING" \
  -lmm 4 -n $i -o psyma_subset_LD_ulmm_trait_$i
  echo "For psyma, ran trait_$i through GEMMA"

  gemma -bfile psymb_subset_LD_plink -p phenos_subset.fam -k ./output/psymb_subset_LD_ksmat.sXX.txt \
  -maf $MIN_MAF -miss "$MAX_MISSING" \
  -lmm 4 -n $i -o psymb_subset_LD_ulmm_trait_$i
  echo "For psymb, ran trait_$i through GEMMA" ;
 done

## adds a columnn to keep track of trait 
for i in $(seq 2 $nPHENOS); # 2 through 4, or however many numbers of traits in $PHENOTYPES
 do
  awk -v i="$i" '{if (NR!=1) print $0, "\t trait_"i}' output/chr_subset_LD_ulmm_trait_$i.assoc.txt | sed '1i chr\tldg\tps\tn_miss\tallele1\tallele0\taf\tbeta\tse\tlogl_H1\tl_remle\tl_mle\tp_wald\tp_lrt\tp_score\trun' > output/chr_subset_LD_ulmm_trait_$i_2.assoc.txt
  awk -v i="$i" '{if (NR!=1) print $0, "\t trait_"i}' output/psyma_subset_LD_ulmm_trait_$i.assoc.txt | sed '1i chr\tldg\tps\tn_miss\tallele1\tallele0\taf\tbeta\tse\tlogl_H1\tl_remle\tl_mle\tp_wald\tp_lrt\tp_score\trun' > output/psyma_subset_LD_ulmm_trait_$i_2.assoc.txt
  awk -v i="$i" '{if (NR!=1) print $0, "\t trait_"i}' output/psymb_subset_LD_ulmm_trait_$i.assoc.txt | sed '1i chr\tldg\tps\tn_miss\tallele1\tallele0\taf\tbeta\tse\tlogl_H1\tl_remle\tl_mle\tp_wald\tp_lrt\tp_score\trun' > output/psymb_subset_LD_ulmm_trait_$i_2.assoc.txt
  mv output/chr_subset_LD_ulmm_trait_$i_2.assoc.txt output/chr_subset_LD_ulmm_trait_$i.assoc.txt
  mv output/psyma_subset_LD_ulmm_trait_$i_2.assoc.txt output/psyma_subset_LD_ulmm_trait_$i.assoc.txt
  mv output/psymb_subset_LD_ulmm_trait_$i_2.assoc.txt output/psymb_subset_LD_ulmm_trait_$i.assoc.txt
 done

## This renames "trait_2, trait_3, etc." to the corresponding name within the phenotype file begining at column 2. 
for str in ${pheno_names[@]}; do
 echo $str
done

for i in $(seq 2 $nPHENOS); # or however many numbers of columns (i.e., traits) 
 do
  sed -i 's/ trait_'"${i}"'/'"${pheno_names[${i}-1]}"'/g' output/chr_subset_LD_ulmm_trait_$i.assoc.txt
  sed -i 's/ trait_'"${i}"'/'"${pheno_names[${i}-1]}"'/g' output/psyma_subset_LD_ulmm_trait_$i.assoc.txt
  sed -i 's/ trait_'"${i}"'/'"${pheno_names[${i}-1]}"'/g' output/psymb_subset_LD_ulmm_trait_$i.assoc.txt
 done

 #### annotate #### 
rm -r annotation/
mkdir -p "annotation"

for i in $(seq 2 $nPHENOS); 
    do
        python merge.py "$chr_var" "$ANNOTATION_FOR_VARIANTS" "output/chr_subset_LD_ulmm_trait_$i.assoc.txt"
        python merge.py "$psyma_var" "$ANNOTATION_FOR_VARIANTS" "output/psyma_subset_LD_ulmm_trait_$i.assoc.txt"  
        python merge.py "$psymb_var" "$ANNOTATION_FOR_VARIANTS" "output/psymb_subset_LD_ulmm_trait_$i.assoc.txt"  
    done                                           


#rename annotation files
for i in $(seq 2 $nPHENOS); # or however many numbers of columns (i.e., traits) 
 do
  mv annotation/merged_chr_subset_LD_ulmm_trait_$i.assoc.txt annotation/merged_chr_subset_LD_ulmm_trait_${pheno_names[${i}-1]}.tsv
  mv annotation/merged_psyma_subset_LD_ulmm_trait_$i.assoc.txt annotation/merged_psyma_subset_LD_ulmm_trait_${pheno_names[${i}-1]}.tsv
  mv annotation/merged_psymb_subset_LD_ulmm_trait_$i.assoc.txt annotation/merged_psymb_subset_LD_ulmm_trait_${pheno_names[${i}-1]}.tsv
 done

# #rename output files
for i in $(seq 2 $nPHENOS); # or however many numbers of columns (i.e., traits) 
 do
  mv output/chr_subset_LD_ulmm_trait_$i.assoc.txt output/chr_subset_LD_ulmm_${pheno_names[${i}-1]}.assoc.txt
  mv output/psyma_subset_LD_ulmm_trait_$i.assoc.txt output/psyma_subset_LD_ulmm_${pheno_names[${i}-1]}.assoc.txt
  mv output/psymb_subset_LD_ulmm_trait_$i.assoc.txt output/psymb_subset_LD_ulmm_${pheno_names[${i}-1]}.assoc.txt
  mv output/chr_subset_LD_ulmm_trait_$i.log.txt output/chr_subset_LD_ulmm_${pheno_names[${i}-1]}.log.txt
  mv output/psyma_subset_LD_ulmm_trait_$i.log.txt output/psyma_subset_LD_ulmm_${pheno_names[${i}-1]}.log.txt
  mv output/psymb_subset_LD_ulmm_trait_$i.log.txt output/psymb_subset_LD_ulmm_${pheno_names[${i}-1]}.log.txt
 done


awk '(NR == 1) || (FNR > 1)' output/*assoc.txt > output/all_assoc.txt
```
